packages/experimental/navigation/components/NativeNavigationTracker.tsx

```diff
@@ -0,0 +1,37 @@
+import React, {ReactNode, forwardRef} from 'react';
+import {TracerProvider} from '@opentelemetry/api';
+
+import useTrace from '../otel/hooks/useTrace';
+import useNativeNavigationTracker, {
+  type NativeNavRef,
+} from '../hooks/useNativeNavigationTracker';
+
+type NativeNavigationTrackerRef = NativeNavRef;
+type NativeNavigationTrackerProps = {
+  children: ReactNode;
+  // selected provider, should be configured by the app consumer
+  provider: TracerProvider;
+};
+
+const NativeNavigationTracker = forwardRef<
+  NativeNavigationTrackerRef,
+  NativeNavigationTrackerProps
+>(({children, provider}, ref) => {
+  // Initializing a Trace instance
+  const tracer = useTrace({name: 'navigation', version: '1.0.0'}, provider);
+
+  /**
+   * NOTE
+   * - should we check if the reference is the one we need even when typescript would complain if it is different?
+   * (People can force the type deactivating the typescript checks)
+   * - should introduce a log to warn the user if the ref is not the one we are expecting and return?
+   */
+
+  useNativeNavigationTracker(ref, tracer);
+
+  return <>{children}</>;
```

Comment:
would just `return children` be equivalent?

Reply:
Typescript complains if I do not add the fragment as wrapper here. It seems like the typescript version doesn't work well with the types from the current `react-native` version (it's a very old one, v0.56.0).